 cameraservice: device1: Fix "slow" legacy HALs

Legacy camera HALs, such as the OnePlus One HAL take
too much time to initialize, so the first call to this
class will always fail (leading to the flashlight QS
tile being unavailable).

In case of errors, give the HAL one more second and retry.

diff --git a/services/camera/libcameraservice/device1/CameraHardwareInterface.h b/services/camera/libcameraservice/device1/CameraHardwareInterface.h
index 02be033..79519d2 100644
--- a/services/camera/libcameraservice/device1/CameraHardwareInterface.h
+++ b/services/camera/libcameraservice/device1/CameraHardwareInterface.h
@@ -118,6 +118,11 @@ class CameraHardwareInterface : public virtual RefBase {
                                      (hw_device_t **)&mDevice);
         } else {
             rc = module->open(mName.string(), (hw_device_t **)&mDevice);
+            if (rc != OK) {
+                // sleep for a second and retry
+                sleep(1);
+                rc = module->open(mName.string(), (hw_device_t **)&mDevice);
+            }
         }
         if (rc != OK) {
             ALOGE("Could not open camera %s: %d", mName.string(), rc);
